### DISCUSSION TITLE
Creating a keep-alive connection to prevent a new connection for every statement

### DISCUSSION THREAD

**Discussion by schroe-ja** [SeaQL/sea-orm#2418] [Q&A] [answered]
Hi, 

I have several functions to communicate with my database i.e.:

```rust
pub async fn get_images(
    con: &DatabaseConnection
) -> Result<Vec<entities::image::Model>, sea_orm::DbErr>{

    let images: Vec<entities::image::Model> = Image::find().all(con).await?;

    Ok(images)
}
```

```rust
pub async fn get_queries(
    con: &DatabaseConnection
) -> Result<Vec<serde_json::Value>, sea_orm::DbErr> {

    let queries = Query::find().into_json().all(con).await?;

    Ok(queries)
}
```
```rust
// Init actix
    HttpServer::new(move || {
        let auth = HttpAuthentication::with_fn(validator);
        App::new()
            .wrap(Logger::default())
            .app_data(web::Data::new(db.clone())) //etc.
```
However, the logic of `sea_orm` requires to pass the `DatabaseConnection` for each statement. Is it possible to create a keep-alive connection that starts an initial connection after the program starts and keeps it up as long as the program runs? Or is the passing the more secure way and I should accept it like it is?

**Comment by tyt2y3** [comment]
this is the rusty-way of doing things. the dependency is explicitly provided, so the logic is easy to test. it also makes it possible to use multiple databases in the same app.

you can, for example, use `lazy_static` to setup a connect pool globally and get it from there everytime you need it. then you don't have to pass the context around.

**Reply by schroe-ja** [reply]
I'm not very familiar creating a connection pool. Do you have an example?

**Comment by schroe-ja** [comment, marked as answer]
Found a solution: https://github.com/SeaQL/sea-orm/discussions/1645#discussioncomment-6329535
